lib/replaceRef.js

```diff
@@ -14,8 +14,20 @@ module.exports = replaceRef;
 * @param  {string} basePath - The root path where any relative schema file references can be resolved
 * @return {object} The schema object with all schema file referenced replaced with the actual file content.
 */
+var idmap = {}
 function replaceRef(schema, basePath) {
     var ref = schema.$ref;
+    // Look through top level schema definitions and store them by id
+    if (schema.hasOwnProperty('definitions') && Object.getOwnPropertyNames(idmap).length == 0) {
+        for (var def in schema.definitions) {
+           idmap[schema.definitions[def].id] = schema.definitions[def]
+        }
+    }
+    // If the ref is against a defined id, replace it
+    if (idmap.hasOwnProperty(ref)) {
+      return replaceRef(idmap[ref], basePath);
```

Comment:
Depending on when my `typeLinking` change is pulled to master, one of us will need to update this code to address the need to return back referenced schemas.